src/plotter.py

```diff
@@ -274,6 +290,46 @@ def create_nan_graph(self, min_date: str, max_date: str):
             nan_dates = gauge_data[str(gauge)].index[gauge_data[str(gauge)].apply(np.isnan)].strftime("%Y-%m-%d")
 
             for date in nan_dates:
-                nan_graph.add_node(node_for_adding=(gauge, date))
+                nan_graph.add_node(node_for_adding=(gauge, date), color="green")
 
         return nan_graph
+
+    def get_subgraph_by_level_group(self, directed_graph: nx.DiGraph):
+        all_connected_subgraphs = []
+        for nb_nodes in range(2, directed_graph.number_of_nodes()):
+            for SG in (directed_graph.subgraph(selected_nodes) for selected_nodes in
+                       itertools.combinations(directed_graph, nb_nodes)):
+                if nx.is_connected(SG):
+                    print(SG.nodes)
+                    all_connected_subgraphs.append(SG)
+        print(all_connected_subgraphs)
+
+    def color_and_label(self, directed_graph: nx.DiGraph, folder_name: str):
+        colors = [""] * len(self.positions)
+        labels = {}
+        g = open(os.path.join(PROJECT_PATH, "data", "level_groups_fontos.json"))
+        level_groups = json.load(g)
+        for gauge in self.gauges:
```

Comment:
This is technically a part for data handling/engineering. It looks more reasonable to move this part to the flood wave detection part, and store the outcome of this snippet in a generated data file (JSON/CSV). Then this method will only load that file and do its job.